tools/TypeDumper/Program.cs

```diff
@@ -0,0 +1,63 @@
+// MIT License.
+
+using System.Reflection;
+using System.Web;
+using System.Web.UI;
+
+// Load up types to include
+var toInclude = new[]
+{
+    typeof(HttpContext).Assembly,
+    typeof(IHttpHandler).Assembly,
+    typeof(Page).Assembly,
+}
+.SelectMany(a => a.GetTypes())
+.Where(t => t.IsPublic)
+.Select(t => t.FullName)
+.ToHashSet();
+
+var dir = Path.Combine(AppContext.BaseDirectory, "netfx");
+using var frameworkContext = new MetadataLoadContext(new PathAssemblyResolver(Directory.EnumerateFiles(dir, "*.dll")));
+
+var forwardedTypes = frameworkContext.LoadFromAssemblyName("System.Web")
+    .ExportedTypes
+    .Select(t => t.FullName)
+    .Where(toInclude.Contains)
+    .OrderBy(t => t);
+
+var file = Path.Combine(GetRoot(), "src", "Shim", "Forwards.cs");
+
+using var fs = File.OpenWrite(file);
+fs.SetLength(0);
+using var writer = new StreamWriter(fs);
+
+writer.WriteLine("// MIT");
```

Comment:
ohh nice... but have we ported all ?

Reply:
nope, this will generate everything we have ported that exists in System.Web.dll and set up type forwarders. We can rerun it periodically (it doesn't format generic types correctly, so they can be manually updated after running the tool)